### ISSUE TITLE
nice: true is ignored on scale updates (initiated by e.g., signals)

### ISSUE THREAD

**Issue by Jerrythafast** [vega/vega#463]
Hey guys,

Got one for you again. As you will notice, the MWE is mostly a copy-paste of the `area` example spec.
```json
{
  "width": 500,
  "height": 200,
  "signals": [
    {"name": "max_x", "init": 20}
  ],
  "data": [
    {
      "name": "table",
      "values": [
        {"x": 0,  "y": 28}, {"x": 2,  "y": 55},
        {"x": 3,  "y": 43}, {"x": 4,  "y": 91},
        {"x": 5,  "y": 81}, {"x": 6,  "y": 53},
        {"x": 7,  "y": 19}, {"x": 8,  "y": 87},
        {"x": 9,  "y": 52}, {"x": 10, "y": 48},
        {"x": 11, "y": 24}, {"x": 12, "y": 49},
        {"x": 13, "y": 87}, {"x": 14, "y": 66},
        {"x": 15, "y": 17}, {"x": 16, "y": 27},
        {"x": 17, "y": 68}, {"x": 18, "y": 16},
        {"x": 19, "y": 49}, {"x": 20, "y": 15}
      ]
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "range": "width",
      "nice": true,
      "domainMin": {"data": "table", "field": "x"},
      "domainMax": {"signal": "max_x"}
    },
    {
      "name": "y",
      "type": "linear",
      "range": "height",
      "nice": true,
      "domain": {"data": "table", "field": "y"}
    }
  ],
  "axes": [
    {"type": "x", "scale": "x"},
    {"type": "y", "scale": "y"}
  ],
  "marks": [
    {
      "type": "area",
      "from": {"data": "table"},
      "properties": {
        "update": {
          "interpolate": {"value": "monotone"},
          "x": {"scale": "x", "field": "x"},
          "y": {"scale": "y", "field": "y"},
          "y2": {"scale": "y", "value": 0},
          "fill": {"value": "steelblue"},
          "fillOpacity": {"value": 1}
        }
      }
    }
  ]
}
```
I would like to draw attention to the horizontal axis. When you first load the spec in the [editor](http://vega.github.io/vega-editor/), the range is `[0, 20]`. If you would load the MWE with `max_x` set to `20.5` initially, the range you get is `[0, 22]` because `nice: true` is specified for the corresponding scale.

However, if you load the MWE with `max_x` set to `20` and then do `ved.view.signal("max_x", 20.5")`, the range changes to `[0, 20.5]`. Which means it is not taking into account the `nice: true` anymore.

Kind regards,
Jerry

**Comment by arvind** [comment]
Thanks Jerry. Fixed with ccc8edb.

**Comment by Jerrythafast** [comment]
Thank you again for the quick fix! And congratulations on Vega's 100 dozen commits :) I think that is a real landmark.
